x-pack/platform/plugins/shared/alerting/server/application/rule_template/methods/find/find_rule_templates.test.ts

```diff
@@ -0,0 +1,486 @@
+
+  test('applies ascending sort order', async () => {
+    unsecuredSavedObjectsClient.find.mockResolvedValueOnce({
+      total: 1,
+      per_page: 10,
+      page: 1,
+      saved_objects: [mockTemplate1],
+    });
+
+    await findRuleTemplates(rulesClientContext, {
+      perPage: 10,
+      page: 1,
+      sortField: 'name',
+      sortOrder: 'asc',
+    });
+
+    expect(unsecuredSavedObjectsClient.find).toHaveBeenCalledWith(
+      expect.objectContaining({
+        sortField: 'name.keyword',
+        sortOrder: 'asc',
+      })
+    );
+  });
+
+  test('handles errors from saved objects client', async () => {
+    const error = new Error('Something went wrong');
+    unsecuredSavedObjectsClient.find.mockRejectedValueOnce(error);
+
+    await expect(
+      findRuleTemplates(rulesClientContext, {
+        perPage: 10,
+        page: 1,
+      })
+    ).rejects.toThrow('Something went wrong');
+  });
+
+  test('returns empty results when no templates found', async () => {
+    unsecuredSavedObjectsClient.find.mockResolvedValueOnce({
+      total: 0,
+      per_page: 10,
+      page: 1,
+      saved_objects: [],
+    });
+
+    const result = await findRuleTemplates(rulesClientContext, {
+      perPage: 10,
+      page: 1,
+    });
+
+    expect(result).toEqual({
+      page: 1,
+      perPage: 10,
+      total: 0,
+      data: [],
+    });
+  });
+
+  describe('authorization', () => {
+    test('throws 403 when user has no access to any rule types', async () => {
+      authorization.getAllAuthorizedRuleTypesFindOperation.mockResolvedValue(new Map());
+
+      await expect(
+        findRuleTemplates(rulesClientContext, {
+          perPage: 10,
+          page: 1,
+        })
+      ).rejects.toMatchObject({
+        output: {
+          statusCode: 403,
+        },
+        message: 'Unauthorized to find rules for any rule types',
+      });
+
+      expect(unsecuredSavedObjectsClient.find).not.toHaveBeenCalled();
+    });
+
+    test('filters templates to only authorized rule types', async () => {
+      // User only has access to test.rule.type
+      authorization.getAllAuthorizedRuleTypesFindOperation.mockResolvedValue(
+        new Map([
+          ['test.rule.type', { authorizedConsumers: { alerts: { read: true, all: true } } }],
+        ])
+      );
+
+      unsecuredSavedObjectsClient.find.mockResolvedValueOnce({
+        total: 1,
+        per_page: 10,
+        page: 1,
+        saved_objects: [mockTemplate1],
+      });
+
+      const result = await findRuleTemplates(rulesClientContext, {
+        perPage: 10,
+        page: 1,
+      });
+
+      expect(result.data).toHaveLength(1);
+      expect(result.data[0].ruleTypeId).toBe('test.rule.type');
+
+      // Verify authorization filter was applied
+      const findCall = unsecuredSavedObjectsClient.find.mock.calls[0][0];
+      expect(findCall.filter).toBeDefined();
+    });
+
+    test('allows templates if user has access to rule type via any consumer', async () => {
```

Comment:
nit: As we mock all responses from the authorization class and we do not have any specific logic about the consumer in the method, maybe we can remove this test.